src/network/message_compact_blocks.rs

```diff
@@ -0,0 +1,39 @@
+//!
+//! BIP152  Compact Blocks network messages
+//!
+
+use util::bip152;
+
+/// sendcmpct message
+#[derive(PartialEq, Eq, Clone, Debug)]
+pub struct SendCmpct {
+    /// Request to be send compact blocks.
+    pub send_compact: bool,
+    /// Compact Blocks protocol version number.
+    pub version: u32,
+}
+impl_consensus_encoding!(SendCmpct, send_compact, version);
+
+/// cmpctblock message
```

Comment:
Can you add add a warning to the effect of "note that the rules for validation before relay for compact blocks is different from headers and regular block messages. Thus, you cannot use compact blocks when relying on an upstream full node to have validated data being forwarded to you".

Reply:
Added.